src/global.d.ts

```diff
@@ -49,6 +49,7 @@ interface Level {
         x: number;
         y: number;
     };
+    timer?: number;
```

Comment:
Let's improve the level interface and add an ending scenario field:

```
interface ExitEndingScenario {
  name: 'exit';
  position: {
    x: number;
    y: number;
  }
}

interface EnemyEndingScenario {
  name: 'enemy';
}

interface TimerEndingScenario {
  name: 'timer';
  timer: number;
}

interface Level {
  // ..
  endingScenario: ExitEndingScenario | EnemyEndingScenario | TimerEndingScenario;
  // ..
}
```
Then let's modify all related codebases with this new field

Reply:
done with it